### PR TITLE
Add classes

### PR DESCRIPTION
**Pull request by wacoo**
- Add person class, its attributes and methods
- Add student class, its attributes and methods
- Add teacher class, its attributes and methods

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
+require('securerandom')
+class Person
+  attr_accessor :name, :age
+  attr_reader :id
+
+  def initialize(name = 'Unknown', age = nil, parent_permission = nil)
+    @id = SecureRandom.uuid
+    @name = name
```
Comment (iambenkis) [inline person.rb]: - [ ] Good job setting default value for optional variable 👍 , As for the `age` variable , it should not be defined with a default value as it is not a optional variable . Note that default value should defined as  "Unknown" for name and true for parent_permission  , Kindly fix that 👍 

Capture

```ruby
class Person
.
.
.
def initialize(age, name = "Unknown", parent_permission = true)
    @id = Random.rand(1..1000)
    @age = age
    @name = name
    @parent_permission = parent_permission
  end
.
.
.
end
```

File: student.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+class Student < Person
+  def initialize(name, age, parent_permission, classroom)
+    super(name, age, parent_permission)
+    @classroom = classroom
+  end
+
```
Comment (iambenkis) [inline student.rb]: - [ ] Student class is well created , however , it should have the optional (or default) value for `name` and for `parent_permission` parameters as describe to in the parent class, Kindly add them 👍 

```ruby
class  Student < Person
 .
 .
 .
  def initialize(age, classroom, name = "Unknown", parent_permission = true)
       #...
  end
 .
 .
 .
end
```

File: teacher.rb [hunk 3]
```diff
@@ -0,0 +1,10 @@
+class Teacher < Person
+  def initialize(name, age, parent_permission, specialization)
+    super(name, age, parent_permission)
+    @specialization = specialization
```
Comment (iambenkis) [inline teacher.rb]: - [ ] Same thing for the Teacher class  which is well created , but  the optional (or default) value for `name` and for `parent_permission` parameters are not defined, Kindly add them 👍 

```ruby
class Teacher < Person
 .
 .
 .
  def initialize(age, specialization, name = "Unknown", parent_permission = true)
       #...
  end
 .
 .
 .
end
```

**Review by iambenkis — CHANGES_REQUESTED** [review]
#### Hi @wacoo  

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation but you are almost there!
You did a great job, keep up the good work 👍🏼

giphy


## Highlights

✔️ Person class added
✔️ Student class added
✔️ Teacher class added

## Suggested changes
_Check the comments under the review._

You can use as many of my suggestions as you want. If there is anything you would like to skip - feel free to do that. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

File: person.rb [hunk 4]
```diff
@@ -0,0 +1,26 @@
+require('securerandom')
+class Person
+  attr_accessor :name, :age
+  attr_reader :id
+
+  def initialize(age, name = 'Unknown', parent_permission = nil)
```
Comment (FrederickMih) [inline person.rb]: - [ ] Hi @wacoo, please let's check the previous reviewer comment once more. He even use a screenshot from the requirements showing that the default value for the parent_permission parameter is `true` and not `nil`. Please let's adhere to the requirements.

File: student.rb [hunk 5]
```diff
@@ -0,0 +1,10 @@
+class Student < Person
+  def initialize(classroom, age, name = 'Unknown', parent_permission = nil)
```
Comment (FrederickMih) [inline student.rb]: - [ ] You did an excellent job by setting the default value for the name parameter to "Unkown" as requested. Unfortunately you forgot to change `nil` to `true`. Working once more on this, will ensure that you'll always abide and work based on the requirements provided by your client in future :smile:

File: teacher.rb [hunk 6]
```diff
@@ -0,0 +1,10 @@
+class Teacher < Person
+  def initialize(specialization, age, name = 'Unknown', parent_permission = nil)
```
Comment (FrederickMih) [inline teacher.rb]: - [ ] Please check once more the optional value for the parent_permission parameter. It takes a boolean value(`true`) as the default value :+1:

**Review by FrederickMih — CHANGES_REQUESTED** [review]
## Hi :raising_hand: :raising_hand: @wacoo , 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


:tada: :champagne:  :tada: :champagne: 

## Highlights :green_circle: 

- Default value for the optional name param was modified as requested :+1: 

## Optional suggestions
**N/A**

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by TedLivist — APPROVED** [review]
## STATUS: APPROVED ✔️ ✔️

Hi @wacoo,

✔️ Implementation of classes in different files based on convention

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**